plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSource.java

```diff
@@ -161,6 +165,9 @@ public Page getNextPage()
             if (dataPage == null) {
                 return null;
             }
+            if (projectionsAdapter.isPresent()) {
+                dataPage = projectionsAdapter.get().adaptPage(dataPage);
+            }
```

Comment:
I think this is right, but just pointed out that the ordering here implies that the `prefilledBlocks` indexes are based on the column ordering _after_ projections.

If you don't have a test with something like `SELECT row.field1, partition_column, row.field2 FROM ...` it'd be good to have.